Assembler.Base/FramesAssembler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+
+namespace Assembler.Base
+{
+    public class FramesAssembler<TFrame> : IAssembler<TFrame>
+        where TFrame : BaseFrame
+    {
+        private readonly IResolver<AssemblingPosition, IFrameHandler<TFrame>> _typeToHandlerResolver;
+        private readonly ILogger _logger;
+
+        public FramesAssembler(IResolver<AssemblingPosition, IFrameHandler<TFrame>> typeToHandlerResolver,
+            ILoggerFactory loggerFactory)
+        {
+            _typeToHandlerResolver = typeToHandlerResolver;
+            _logger = loggerFactory.GetLogger(this);
+        }
+
+        public void Assemble(TFrame frame)
+        {
+            try
+            {
+                IFrameHandler<TFrame> handler = _typeToHandlerResolver.Resolve(frame.AssemblingPosition);
+
+                _logger.Debug($"Frame [{frame.Guid}] is being sent to the handler.");
+
+                handler.Handle(frame);
+            }
+            catch (KeyNotFoundException e)
+            {
+                _logger.Error($"No matching resolver was found for the message [{frame.Guid}]," +
+                              $"It won't be used in the assembling process. \n {e}");
+            }
```

Comment:
That's the only exception that the resolver throws

Reply:
1. this is abstraction, so you cant know
2. the Handler.Handle is also in the try catch (https://github.com/reverser-gever/Assembler/pull/2#discussion_r400371251)
3. Generally, if you dont treat different in various kinds of exceptions, there is no reason to catch specific one (unless you desire some kinds of exceptions to be thrown to the higher level).

Bad catch example:
![image](https://user-images.githubusercontent.com/62211836/78179338-b547ef80-7469-11ea-9d71-1cf41a8fd3b8.png)
thats bad because it can (and should) be combined to this:
![image](https://user-images.githubusercontent.com/62211836/78179435-df99ad00-7469-11ea-9979-10f31fbb5bd0.png)


Good catch example:
![image](https://user-images.githubusercontent.com/62211836/78179542-0eb01e80-746a-11ea-9bae-0f0b3b5495ed.png)

Reply:
f